test/precompile.jl

```diff
@@ -630,10 +630,10 @@ precompile_test_harness("code caching") do dir
     Base.invokelatest() do
         Dict{M.X2,Any}()[M.X2()] = nothing
     end
-    @test M.X2 ∈ m.roots
+    @test_broken M.X2 ∈ m.roots
```

Comment:
Why mark these as broken? Presumably, they need to be fixed before the PR is merged but marking them as broken will make it look like CI passed.

Reply:
Yes, it should be fixed before status changed from draft. So far, I'm kind of lost because it tests if the method is rooted for caching external CodeInstances. (Comes from #43793)

Reply:
That's just a convenience, it's not really a test about Dict per se. If you're changing the implementation of Dict, feel free to change this test so that it tests root-insertion on other code.

Reply:
You could probably just create `OldDict` that's the current implementation and use it for the purpose of this test.